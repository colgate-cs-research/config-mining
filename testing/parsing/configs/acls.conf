interface GigabitEthernet0/1
    ip address 10.0.1.1 255.255.255.0
!
interface GigabitEthernet0/2
    ip address 10.0.2.1 255.255.255.0
    ip access-group aclA in
!
interface GigabitEthernet0/3
    ip address 10.0.3.1 255.255.255.0
    ip access-group aclB out
!
interface GigabitEthernet0/4
    ip address 10.0.4.1 255.255.255.0
    ip access-group aclC in
    ip access-group aclD out
!
ip access-list standard aclA
    deny 20.0.1.128 0.0.0.127
    permit 20.0.1.0 0.0.0.255
!
ip access-list standard aclB
    deny 20.0.2.1
    permit any
!
ip access-list extended aclC
    remark Test filtering by source or destination
    permit ip 20.0.3.0 0.0.0.255 any
    permit ip host 20.0.4.1 any
    deny ip any 20.0.5.0 0.0.0.255
    deny ip any host 20.0.6.1
    remark Test filtering by source and destination
    permit ip 20.0.7.0 0.0.0.255 20.0.8.0 0.0.0.255
    permit ip host 20.0.9.1 host 20.0.10.1
    permit ip 20.0.11.0 0.0.0.255 host 20.0.12.1
    permit ip host 20.0.13.1 20.0.14.0 0.0.0.255
!
ip access-list extended aclD
    remark Test filtering based on different protocols
    permit ip 20.0.15.0 0.0.0.255 any eq bootp
    permit tcp 20.0.16.0 0.0.0.255 any
    permit udp 20.0.17.0 0.0.0.255 any
!

