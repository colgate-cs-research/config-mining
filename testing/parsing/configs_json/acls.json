{
    "acls": {
        "aclA": {
            "lines": [
                {
                    "action": "DENY",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "20.0.1.128/25"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "20.0.1.0/24"
                }
            ],
            "name": "aclA",
            "remarks": []
        },
        "aclB": {
            "lines": [
                {
                    "action": "DENY",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "20.0.2.1"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "0.0.0.0/0"
                }
            ],
            "name": "aclB",
            "remarks": []
        },
        "aclC": {
            "lines": [
                {
                    "action": "PERMIT",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "20.0.3.0/24"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "20.0.4.1"
                },
                {
                    "action": "DENY",
                    "dstIps": "20.0.5.0/24",
                    "srcIps": "0.0.0.0/0"
                },
                {
                    "action": "DENY",
                    "dstIps": "20.0.6.1",
                    "srcIps": "0.0.0.0/0"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "20.0.8.0/24",
                    "srcIps": "20.0.7.0/24"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "20.0.10.1",
                    "srcIps": "20.0.9.1"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "20.0.12.1",
                    "srcIps": "20.0.11.0/24"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "20.0.14.0/24",
                    "srcIps": "20.0.13.1"
                }
            ],
            "name": "aclC",
            "remarks": [
                "Test filtering by source or destination",
                "Test filtering by source and destination"
            ]
        },
        "aclD": {
            "lines": [
                {
                    "action": "PERMIT",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "20.0.16.0/24"
                },
                {
                    "action": "PERMIT",
                    "dstIps": "0.0.0.0/0",
                    "srcIps": "20.0.17.0/24"
                }
            ],
            "name": "aclD",
            "remarks": [
                "Test filtering based on different protocols"
            ]
        }
    },
    "interfaces": {
        "GigabitEthernet0/1": {
            "access_vlan": null,
            "address": "10.0.1.1/24",
            "allowed_vlans": null,
            "description": null,
            "in_acl": null,
            "name": "GigabitEthernet0/1",
            "out_acl": null,
            "switchport": null
        },
        "GigabitEthernet0/2": {
            "access_vlan": null,
            "address": "10.0.2.1/24",
            "allowed_vlans": null,
            "description": null,
            "in_acl": "aclA",
            "name": "GigabitEthernet0/2",
            "out_acl": null,
            "switchport": null
        },
        "GigabitEthernet0/3": {
            "access_vlan": null,
            "address": "10.0.3.1/24",
            "allowed_vlans": null,
            "description": null,
            "in_acl": null,
            "name": "GigabitEthernet0/3",
            "out_acl": "aclB",
            "switchport": null
        },
        "GigabitEthernet0/4": {
            "access_vlan": null,
            "address": "10.0.4.1/24",
            "allowed_vlans": null,
            "description": null,
            "in_acl": "aclC",
            "name": "GigabitEthernet0/4",
            "out_acl": "aclD",
            "switchport": null
        },
        "GigabitEthernet0/5": {
            "access_vlan": null,
            "address": "10.0.5.1/24",
            "allowed_vlans": null,
            "description": null,
            "in_acl": "aclC",
            "name": "GigabitEthernet0/5",
            "out_acl": "aclB",
            "switchport": null
        },
        "GigabitEthernet0/6": {
            "access_vlan": null,
            "address": "20.0.2.1/24",
            "allowed_vlans": null,
            "description": null,
            "in_acl": null,
            "name": "GigabitEthernet0/6",
            "out_acl": "aclB",
            "switchport": null
        }
    },
    "name": "acls",
    "vlans": {}
}