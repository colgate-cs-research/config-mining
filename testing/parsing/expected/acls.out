C(Interface -> Have ACL references)
	Interfaces with ACL reference(s): 3
	Support (num interfaces): 4
	Confidence: 0.75

C('in' access list -> 'out' access list)
	Interfaces with 'out' access lists: 1
	Support (num interfaces with 'in' access list): 3
	Confidence: 0.3333333333333333

C(ACL covers interfaces IP -> interface has that ACL applied
	Two way ACL-Interface references: 0
	Support (num IP addresses covered in ACL): 0
IToACL
{
    "GigabitEthernet0/2": [
        "aclA"
    ],
    "GigabitEthernet0/3": [
        "aclB"
    ],
    "GigabitEthernet0/4": [
        "aclC",
        "aclD"
    ]
}
------------------------------------------------
interfaceIP
{
    "10.0.2.1": [
        "aclA"
    ],
    "10.0.3.1": [
        "aclB"
    ],
    "10.0.4.1": [
        "aclC",
        "aclD"
    ]
}
------------------------------------------------
ACLtoI
{
    "aclA": [
        [
            "20.0.1.128",
            "0.0.0.127"
        ],
        [
            "20.0.1.0",
            "0.0.0.255"
        ]
    ],
    "aclB": [
        [
            "20.0.2.1",
            "0.0.0.0"
        ]
    ],
    "aclC": [
        [
            "20.0.3.0",
            "0.0.0.255"
        ],
        [
            "20.0.4.1",
            "0.0.0.0"
        ],
        [
            "20.0.5.0",
            "0.0.0.255"
        ],
        [
            "20.0.6.1",
            "0.0.0.0"
        ],
        [
            "20.0.7.0",
            "0.0.0.255"
        ],
        [
            "20.0.8.0",
            "0.0.0.255"
        ],
        [
            "20.0.9.1",
            "0.0.0.0"
        ],
        [
            "20.0.10.1",
            "0.0.0.0"
        ],
        [
            "20.0.11.0",
            "0.0.0.255"
        ],
        [
            "20.0.12.1",
            "0.0.0.0"
        ],
        [
            "20.0.13.1",
            "0.0.0.0"
        ],
        [
            "20.0.14.0",
            "0.0.0.255"
        ]
    ],
    "aclD": [
        [
            "20.0.16.0",
            "0.0.0.255"
        ],
        [
            "20.0.17.0",
            "0.0.0.255"
        ]
    ]
}
------------------------------------------------
